filesmanager/filesmanager.py

```diff
@@ -48,20 +132,339 @@ def student_view(self, context=None):
+        return frag
+
+    @XBlock.json_handler
+    def get_directories(self, data, suffix=''):
+        """Get the list of directories.
+
+
+        Returns: the list of directories is stored in the XBlock settings.
+
+        [
+            {
+                "name": "Folder 1",
+                "type": "directory",
+                "path": "Folder 1",
+                "metadata": {
+                    "id": ..,
+                    ...
+                },
+                "children": [
+                    ...
+                ]
+            }
+        ]
+        """
+        return {
+            "status": "success",
+            "content": self.directories,
+        }
+
     @XBlock.json_handler
-    def increment_count(self, data, suffix=''):
+    def clear_directories(self, data, suffix=''):
+        """Clear the list of directories without removing files from course assets.
+
+        This method is intended to be used for testing purposes.
+
+        Returns: an empty list of directories.
         """
-        An example handler, which increments the data.
+        self.directories = []
+        self.incremental_directory_id = 0
+        return {
+            "status": "success",
+            "content": self.directories,
+        }
+
+    @XBlock.json_handler
+    def get_content(self, data, suffix=''):
+        """Get the content of a directory.
+
+        Arguments:
+            data: the path of the directory to be retrieved.
+
+        Returns: the content of the directory if found, an empty sequence otherwise.
         """
-        if suffix:
-            pass  # TO-DO: Use the suffix when storing data.
-        # Just to show data coming in...
-        assert data['hello'] == 'world'
+        path = data.get("path")
+        if not path:
+            return {
+                "status": "error",
+                "message": "Path not found",
+            }
+        content, _, _ = self.get_content_by_path(path)
+        return {
+            "status": "success",
+            "content": content,
+        }
+
+    @XBlock.json_handler
+    def add_directory(self, data, suffix=''):
+        """Add a directory to a target directory.
+
+        The new directory will:
+        - Be added to the target directory, if found. Otherwise, an error will be returned.
+        - Be added to the root directory if no target directory is specified.
+        - Have a unique incremental ID.
+        - Have a path composed by the target directory path and the directory name.
+        - Have an empty list of children.
 
-        self.count += 1
-        return {"count": self.count}
+        Arguments:
+            name: the name and path of the directory to be added.
+            path: the path of the target directory where the new directory will be added.
+        """
+        directory_name = data.get("name")
+        path = data.get("path")
+        target_directory = self.get_target_directory(path)
+        if not target_directory:
```

Comment:
I think the `get_target_directory` function should return `None`, in case it does not find the path. And check in this condition explicitly:
```python
if target_directory is None:
```

Reply:
Why would that be better?

Reply:
I got that now. Not sure what I changed to break it. Changes done! Thanks